### PR TITLE
feat(idp): http idleTimeout on social idp

### PR DESCRIPTION
**Pull request by remisultan**
fixes: https://github.com/gravitee-io/issues/issues/3505

Example below with a saved IdP and a new one in (The `idleTimeout` per default is set to 10000 ms)

image
image

### CODE REVIEW

**Comment by brasseld** [comment]
@tcompiegne just asking myself why we need pool size to 200 or 100 ? looks really big to me for the AM use-case.
WDYT?

**Comment by remisultan** [comment]
WIP because i have a few questions:

I was able to determine that `idleTimeout` was the field in the WebClientOption for this feature.
I determined it on my own using a mock http server with idling for 5 seconds then closing the connection on a 10 second request.
I could also determine that field was set by breakpointing my was into the code.

However i was not able to check where this has to be tested within the project itself.
I tried to refer to the field above `connectTimeout` but could not find such files.

I also tried to wrangle with `TwitterAuthenticationProviderTest.class` 
https://github.com/gravitee-io/gravitee-access-management/blob/b8900ae365d900450d2fa43ee88b8a0e2b02e471/gravitee-am-identityprovider/gravitee-am-identityprovider-twitter/src/test/java/io/gravitee/am/identityprovider/twitter/authentication/TwitterAuthenticationProviderTest.java#L79

I tried to by-pass event phase as did there by getting it to `RECEIVE_RESPONSE` phase and put a `Thread.sleep` without success.

This is open to discussion and guidance is very appreciated 🙏

**Comment by remisultan** [comment]
> @tcompiegne just asking myself why we need pool size to 200 or 100 ? looks really big to me for the AM use-case.
> WDYT?

@brasseld interestingly the pool size varies regarding the idp ? 🤔

File: gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/GithubIdentityProviderConfiguration.java [hunk 1]
```diff
@@ -100,6 +101,14 @@ public void setConnectTimeout(Integer connectTimeout) {
         this.connectTimeout = connectTimeout;
     }
 
+    public Integer getIdleTimeout() {
```
Comment (leleueri) [inline gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/GithubIdentityProviderConfiguration.java:104]: The idleTimeout is never set to the WebClientOptions

https://github.com/gravitee-io/gravitee-access-management/blob/issues%2F%233505-http-request-timeout-social-idp/gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/authentication/spring/GithubAuthenticationProviderConfiguration.java#L55

File: gravitee-am-identityprovider/gravitee-am-identityprovider-google/src/main/java/io/gravitee/am/identityprovider/google/GoogleIdentityProviderConfiguration.java [hunk 2]
```diff
@@ -88,6 +89,14 @@ public void setConnectTimeout(Integer connectTimeout) {
         this.connectTimeout = connectTimeout;
     }
 
+    public Integer getIdleTimeout() {
```
Comment (leleueri) [inline gravitee-am-identityprovider/gravitee-am-identityprovider-google/src/main/java/io/gravitee/am/identityprovider/google/GoogleIdentityProviderConfiguration.java:92]: The idleTimeout is never set to the WebClientOptions

https://github.com/gravitee-io/gravitee-access-management/blob/issues%2F%233505-http-request-timeout-social-idp/gravitee-am-identityprovider/gravitee-am-identityprovider-google/src/main/java/io/gravitee/am/identityprovider/google/authentication/spring/GoogleAuthenticationProviderConfiguration.java#L55

**Review by leleueri — CHANGES_REQUESTED** [review]

File: gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/GithubIdentityProviderConfiguration.java (hunk 1 above)
Comment (remisultan) [reply gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/GithubIdentityProviderConfiguration.java:104]: Changes applied, thanks.

File: gravitee-am-identityprovider/gravitee-am-identityprovider-google/src/main/java/io/gravitee/am/identityprovider/google/GoogleIdentityProviderConfiguration.java (hunk 2 above)
Comment (remisultan) [reply gravitee-am-identityprovider/gravitee-am-identityprovider-google/src/main/java/io/gravitee/am/identityprovider/google/GoogleIdentityProviderConfiguration.java:92]: Changes applied, thanks.

Comment (remisultan) [reply gravitee-am-identityprovider/gravitee-am-identityprovider-google/src/main/java/io/gravitee/am/identityprovider/google/GoogleIdentityProviderConfiguration.java:92]: Done

File: gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/GithubIdentityProviderConfiguration.java (hunk 1 above)
Comment (remisultan) [reply gravitee-am-identityprovider/gravitee-am-identityprovider-github/src/main/java/io/gravitee/am/identityprovider/github/GithubIdentityProviderConfiguration.java:104]: Done

**Review by leleueri — APPROVED** [review]

**Pull request merged**